lib/fake_ifdh.py

```diff
@@ -0,0 +1,155 @@
+#
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+import os
+import time
+import argparse
+
+VAULT_HOST = "fermicloud543.fnal.gov"
+DEFAULT_ROLE = "Analysis"
+
+
+def getTmp():
+    return os.environ.get("TMPDIR", "/tmp")
+
+
+def getExp():
+    for ev in ["GROUP", "EXPERIMENT", "SAM_EXPERIMENT"]:
+        if os.environ.get(ev, None):
+            return os.environ.get(ev)
+    # otherwise guess primary group...
+    f = os.popen("id -gn", "r")
+    exp = f.read()
+    f.close()
+    return exp
+
+
+def getRole(role_override=None):
+    if role_override:
+        return role_override
+    elif os.environ["USER"][-3:] == "pro":
+        return "Production"
+    else:
+        return DEFAULT_ROLE 
+
+
+def checkToken(tokenfile):
+    f = os.popen("decode_token.sh -e exp %s 2>/dev/null" % tokenfile, "r")
+    exp_time = f.read()
+    f.close()
+    return  exp_time and ((int(exp_time) - time.time()) > 60)
+
+def getToken(role=DEFAULT_ROLE):
+    pid = os.getuid()
+    tmp = getTmp()
+    exp = getExp()
+    if exp == "samdev":
+        issuer = "fermilab"
+    else:
+        issuer = exp
+    tokenfile = "%s/bt_token_%s_%s_%s" % (tmp, issuer, role, pid)
+    os.environ["BEARER_TOKEN_FILE"] = tokenfile
+    if not checkToken(tokenfile):
+        cmd = "htgettoken -a %s -i %s" % (VAULT_HOST, issuer)
+        if role != DEFAULT_ROLE:
+            cmd = "%s -r %s" % (cmd, role.lower()) # Token-world wants all-lower
+        # send htgettoken output to stderr because invokers read stdout
+        res = os.system("%s >&2" % cmd)
+        if res != 0:
+            raise PermissionError("Failed attempting '%s'" % cmd)
+        if checkToken(tokenfile):
+            return(tokenfile)
+        else:
+            raise PermissionError("Failed attempting '%s'" % cmd)
+    else:
+        return(tokenfile)
+
+
+def getProxy(role=DEFAULT_ROLE):
+    pid = os.getuid()
+    tmp = getTmp()
+    exp = getExp()
+    certfile = "%s/x509up_u%s" % (tmp, pid)
+    if exp == "samdev":
+        issuer = "fermilab"
+        igroup = "fermilab"
+    elif exp in ("lsst", "dune", "fermilab", "des"):
+        issuer = exp
+        igroup = exp
+    else:
+        issuer = "fermilab"
+        igroup = "fermilab/" + exp
+    vomsfile = "%s/x509up_%s_%s_%s" % (tmp, exp, role, pid)
+    chk_cmd = "voms-proxy-info -exists -valid 0:10 -file %s" % vomsfile
+    if 0 != os.system(chk_cmd):
+        cmd = "cigetcert -i 'Fermi National Accelerator Laboratory' -n -o %s" % certfile
+        # send htgettoken output to stderr because invokers read stdout
```

Comment:
Since this is no longer true, we should take out the comment.

Reply:
Well, I'm torn.   I'm trying to emulate ifdh, and that is its rule... We don't rely on it anymore in this code though... sigh..

Reply:
Lets leave it.

Reply:
Yeah - I misread line 106 (and line 76) - you're sending the output to stderr, not stdout, which is what you said in the comment.  My mistake.